src/libMetrics/Logging.cpp

```diff
@@ -142,25 +142,36 @@ std::shared_ptr<logs::LoggerProvider> InitNoop() {
 
 }  // namespace
 
-Logging::Logging() {
-  std::string cmp(LOGGING_ZILLIQA_PROVIDER);
+void Logging::Initialize(std::string_view identity /*= {}*/,
+                         std::string provider /*= LOGGING_ZILLIQA_PROVIDER*/) {
+  static std::array<std::string_view, 4> allowedIdentities = {
+      "seedprv", "seedpub", "dsguard", "lookup"};
+  if (std::none_of(std::begin(allowedIdentities), std::end(allowedIdentities),
+                   [identity](const auto& allowedIdentity) {
+                     return identity.find(allowedIdentity) != std::string::npos;
+                   })) {
+    provider = "NONE";
+  }
+
+  transform(provider.begin(), provider.end(), provider.begin(), ::tolower);
```

Comment:
Worth a utility function? (or using the boost alg?) - this is really nitpicky!

Reply:
Yeah, changed to use the boost::algorithm::to_lower.